src/Stryker.Core/Stryker.Core/ExcludableString.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Text.RegularExpressions;
+using DotNet.Globbing;
+
+namespace Stryker.Core
+{
+    public readonly struct ExcludableString
+    {
+        private static readonly Regex _mutantSpansRegex = new("(\\{(\\d+)\\.\\.(\\d+)\\})+$");
+
+        public ExcludableString(string s)
```

Comment:
That might still be confusing. Can't we move all the parse logic into this class and call it something like ExclusionPatternParser or something? Currently part of the parsin is moved here but some remains in FilePattern.

Reply:
Yeah, `ExclusionPattern` is better. I didn't like the name but didn't want to say this loud to not bias you :D 

Now, as for moving the other part of the parsing logic here - that would kind of defeat the purpose. Because we want to parse this thing - `{10..20}{20..30}` - differently for F# and C#. Different parsing techniques and different resulting "hard types" at the end (`TextSpan` for C# and `Range` for F#).

Reply:
I meant up until but not including the hard types. The first part is still just string matching right?

Reply:
Ohh right, yeah, that we can do I think.